### DISCUSSION TITLE
VAL3 version

### DISCUSSION THREAD

**Discussion by mzacri** [ros-industrial/staubli_val3_driver#51] [Q&A]
Hi everyone,

I have a Staubli rx130b robot with a CS8 contrôller. The installed VAL3 version is s5.0.1 and the robot CPU is from 2003 era. Does anyone know if the update to s7.x.x is possible or not? If not, is there another way to stream trajectories to CS8 controller (I have read about Real-time Robot Controller Abstraction Layer in the SENSOR-GUIDED ROBOTIC LASER WELDING thesis but I cannot find it)?

Thanks a lot!

**Comment by gavanderhoorn** [comment]
I'm afraid I wouldn't be able to help you with the upgrade question.

re: streaming trajectories: CS8 controllers should still support the *low level interface*, but going that route would mean you'd have to implement everything yourself.

I can't say whether that'd be too much work or not, as it would depend on your expected gains.

There could of course also be other tools that facilitate external control, I wouldn't be able to recommend you one though.
